test/functorch/test_control_flow.py

```diff
@@ -2459,8 +2527,16 @@ def add_scalar_carry(x: torch.Tensor, y: torch.Tensor):
         self.assertEqual(result_init, result_exp)
         self.assertEqual(result_init[0], torch.tensor([3.0], device=device))
 
+        if autograd:
```

Comment:
all the autograd tests looks similar, can we extract them out as a funciton: check_autograd?

Reply:
I factored out most of them